### DISCUSSION TITLE
header options

### DISCUSSION THREAD

**Discussion by dineshreddypaidi** [encode/django-rest-framework#9736] [Question & Answer] [answered]
iam working with token autentication
for some of endpoint need auth token and need to be included in header options..
but in static web browser doesnot show any of header options and for every endpoint iam getting an auth token error
i checked endpoints with other clients postman. it is working fine.. 
can add a feature for posting header option also to the static web browser views

**Comment by Dhruv-Bajaj-code** [comment, marked as answer]
After receiving the token from the backend, it should be saved in the browser's local storage for future authentication, as follows:

```
const authToken = response.headers.get('Authorization');
if (authToken) {
    localStorage.setItem('authToken', authToken);
}
```


Now, with every request, you will need to include the token in the header. You can do it as follows:

```
function fetchWithAuth(url, options = {}) {
    const authToken = localStorage.getItem('authToken'); 

    
    const headers = {
        ...options.headers,
        'Authorization': authToken ? authToken : '',
        'Content-Type': 'application/json', 
    };

    /
    return fetch(url, {
        ...options,
        headers,
    });
}
```
**Both are parts of the frontend.**

please let me know if this worked.
